src/transformers/models/maskformer/configuration_maskformer.py

```diff
@@ -0,0 +1,179 @@
+}
+
+logger = logging.get_logger(__name__)
+
+
+class MaskFormerConfig(PretrainedConfig):
+    r"""
+    This is the configuration class to store the configuration of a [`MaskFormer`]. It is used to instantiate a
+    MaskFormer model according to the specified arguments, defining the model architecture. Instantiating a
+    configuration with the defaults will yield a similar configuration to that of the
+    "Francesco/maskformer-swin-base-ade" architecture trained on ADE20k-150
+
+    Configuration objects inherit from [`PretrainedConfig`] and can be used to control the model outputs. Read the
+    documentation from [`PretrainedConfig`] for more information.
+
+    Currently, maskformer supports only Swin backbone.
+
+    Args:
+        mask_feature_size (`int`, *optional*, defaults to 256):
+            The masks' features size, this value will also be used to specify the Feature Pyramid Network featuresc
+            size.
+        no_object_weight (`float`, *optional*, defaults to 0.1):
+            Weight to apply to the null class .
+        use_auxilary_loss (`bool`, *optional*, defaults to `False`):
+            If `true` [`MaskFormerOutput`] will contain.
+        backbone_config (`Dict`, *optional*):
+            The configuration passed to the backbone, if unset, the configuration corresponding to `swin-base` will be
+            used.
+        detr_config (`Dict`, *optional*):
+            The configuration passed to the Detr model, if unset the base config for `detr` will be used.
+        init_std (`float`, *optional*, defaults to 0.02):
+            The standard deviation of the truncated_normal_initializer for initializing all weight matrices.
+        init_xavier_std (`float`, *optional*, defaults to 1):
+            The scaling factor used for the Xavier initialization gain in the HM Attention map module.
+        dice_weight (`float`, *optional*, defaults to 1.0):
+            The weight for the dice loss.
+        cross_entropy_weight (`float`, *optional*, defaults to 1.0):
+            The weight for the cross entropy loss.
+        mask_weight (`float`, *optional*, defaults to 20.0):
+            The weight for the mask loss.
+        num_labels (`int`, *optional*, defaults to 150):
+            The number of labels.
+
+    Raises:
+        `ValueError`: Raised if the backbone model type selected is not in `["swin"]`
+
+    Examples:
+
+    ```python
+    >>> from transformers import MaskFormerConfig, MaskFormerModel
+
+    >>> # Initializing a MaskFormer facebook/maskformer-swin-base-ade configuration
+    >>> configuration = MaskFormerConfig()
+
+    >>> # Initializing a model from the facebook/maskformer-swin-base-ade style configuration
+    >>> model = MaskFormerModel(configuration)
+
+    >>> # Accessing the model configuration
+    >>> configuration = model.config
+    ```
+
+    """
+    model_type = "maskformer"
+    attribute_map = {"hidden_size": "mask_feature_size"}
```

Comment:
Is there a reason we can't use `hidden_size` right away?

Reply:
`mask_feature_size` is a more appropriate name given the model, but if you think is better to pass `hidden_size` I can definitely change it and explain what it is in the doc

Reply:
Hmm ok, let's keep `mask_feature_size` in that case.